bitcoin/src/crypto/key.rs

```diff
@@ -237,18 +236,18 @@ impl fmt::Display for PublicKey {
 impl FromStr for PublicKey {
     type Err = ParsePublicKeyError;
     fn from_str(s: &str) -> Result<PublicKey, ParsePublicKeyError> {
-        use HexToArrayError::*;
+        use hex_stable::ToArrayError::*;
 
         match s.len() {
             66 => {
-                let bytes = <[u8; 33]>::from_hex(s).map_err(|e| match e {
+                let bytes = hex_stable::decode_array::<33>(s).map_err(|e| match e.parse_error() {
                     InvalidChar(e) => ParsePublicKeyError::InvalidChar(e),
                     InvalidLength(_) => unreachable!("length checked already"),
```

Comment:
Right, so the inner enum is actually used by this code. In which case I'm convinced `HexToArrayError` should be just enum.

Reply:
It was that before. @apoelstra can you come in to bat on this one please.